edx_exams/apps/core/api.py

```diff
@@ -142,3 +143,58 @@ def get_exam_attempt_time_remaining(exam_attempt, now=None):
         secs_to_end = (end_time-now).total_seconds()
 
     return secs_to_end
+
+
+def create_exam_attempt(exam_id, user_id):
+    """
+    Creates an exam attempt for user_id against exam_id. There should only
+    be one exam_attempt per user per exam.
+    """
+
+    exam_obj = Exam.get_exam_by_id(exam_id)
+
+    if exam_obj is None:
+        err_msg = (
+            f'Exam with exam_id={exam_id} does not exist.'
+        )
+        raise ExamDoesNotExist(err_msg)
+
+    log.info(
```

Comment:
Nit: Was there a particular reason you'd wanted to log before and after creating the exam attempt? The reason I ask is because I wonder if it is confusing for logging if there is a log message but then the attempt is not actually created. But then again, I can see that logging could be useful for debugging purposes and I don't think there's any cost to logging so would defer to you here! :)

Reply:
I think there's a tendency to log as much info as is useful for proctoring, because then there is a wealth of information available in splunk when trying to debug something. 

It's useful to log before and after so that it's clear if we tried to create an attempt, and if we were actually successful in creating that attempt.

Reply:
Ah that makes total sense and is a really good point.